lib/trino-filesystem-azure/src/main/java/io/trino/filesystem/azure/AzureFileSystemFactory.java

```diff
@@ -65,14 +71,39 @@ public AzureFileSystemFactory(
         this.maxWriteConcurrency = maxWriteConcurrency;
         this.maxSingleUploadSize = requireNonNull(maxSingleUploadSize, "maxSingleUploadSize is null");
         this.tracingOptions = new OpenTelemetryTracingOptions().setOpenTelemetry(openTelemetry);
-        this.httpClient = HttpClient.createDefault((HttpClientOptions) new HttpClientOptions()
-                .setHttpClientProvider(OkHttpAsyncClientProvider.class)
-                .setTracingOptions(tracingOptions));
+
+        okHttpClient = new OkHttpClient.Builder().build();
+        HttpClientOptions clientOptions = new HttpClientOptions();
+        clientOptions.setTracingOptions(tracingOptions);
+        httpClient = createAzureHttpClient(okHttpClient, clientOptions);
+    }
+
+    @PreDestroy
+    public void destroy()
+    {
+        okHttpClient.dispatcher().executorService().shutdownNow();
+        okHttpClient.connectionPool().evictAll();
     }
 
     @Override
     public TrinoFileSystem create(ConnectorIdentity identity)
     {
         return new AzureFileSystem(httpClient, tracingOptions, auth, readBlockSize, writeBlockSize, maxWriteConcurrency, maxSingleUploadSize);
     }
+
+    public static HttpClient createAzureHttpClient(OkHttpClient okHttpClient, HttpClientOptions clientOptions)
+    {
+        Integer poolSize = clientOptions.getMaximumConnectionPoolSize();
+        // By default, OkHttp uses a maximum idle connection count of 5.
+        int maximumConnectionPoolSize = (poolSize != null && poolSize > 0) ? poolSize : 5;
+
+        return new OkHttpAsyncHttpClientBuilder(okHttpClient)
+                .proxy(clientOptions.getProxyOptions())
+                .configuration(clientOptions.getConfiguration())
+                .connectionTimeout(clientOptions.getConnectTimeout())
+                .writeTimeout(clientOptions.getWriteTimeout())
+                .readTimeout(clientOptions.getReadTimeout())
+                .connectionPool(new ConnectionPool(maximumConnectionPoolSize,
+                        clientOptions.getConnectionIdleTimeout().toMillis(), TimeUnit.MILLISECONDS)).build();
```

Comment:
There may be potential for further config options in `AzureFileSystemConfig`
cc @anusudarsan

Reply:
ya we will have test internally with defaults and see if there is a need for it